compiler/src/iree/compiler/Dialect/Flow/Transforms/FormDispatchRegions.cpp

```diff
@@ -99,14 +99,14 @@ LogicalResult simplifyDimOps(RewriterBase &rewriter,
 
     // Try to simplify dynamic dims.
     SmallVector<Value> dynamicDims;
-    if (failed(IREE::Flow::reifyDynamicResultDims(rewriter, dimOp.getSource(),
-                                                  dynamicDims)))
-      return failure();
-    unsigned ctr = 0;
-    for (int64_t i = 0; i < *dimOp.getConstantIndex(); ++i)
-      if (tensorType.isDynamicDim(i))
-        ++ctr;
-    rewriter.replaceOp(dimOp, dynamicDims[ctr]);
+    if (succeeded(IREE::Flow::getOptimizedDynamicResultDims(
+            rewriter, dimOp.getSource(), dynamicDims))) {
+      unsigned ctr = 0;
+      for (int64_t i = 0; i < *dimOp.getConstantIndex(); ++i)
+        if (tensorType.isDynamicDim(i))
+          ++ctr;
+      rewriter.replaceOp(dimOp, dynamicDims[ctr]);
+    }
   }
```

Comment:
Was anything actually changed here?

Reply:
Yes! The `getOptimizedDynamicResultDims` behaviour differs from `reifyDynamicResultDims` by not generating the `tensor.dim` operations as a fallback. This is what basically caused the exponential issue and was originally added here: https://github.com/openxla/iree/pull/10751/files

The idea here is to stop conflating the concept of "reification" and "optimization", since what the caller here wants to do is optimize an existing `tensor.dim` rather than reify a dynamic dimension.

Reply:
Oh sorry, i didn't notice the function call change there, thanks.